### PR TITLE
Add launch review regression coverage and stabilize daemon tests

### PR DESCRIPTION
**Pull request by AmyShang-alt**
## Summary
- add Playwright regression coverage for launch-review flows across chat, previews, plugin authoring, diagnostics, design systems, automations, and integrations
- harden daemon regression coverage around plugin-authoring completion, diagnostics export, raw HTML routes, AMR model normalization, and terminal state handling
- stabilize daemon runtime tests by isolating CLI resolution env and making `startServer` wait for a real `listening` event before resolving its bound port

## Surface area
- [x] None
- [ ] UI
- [ ] Backend
- [ ] Infra
- [ ] Schema / contract

## Testing
- `pnpm exec playwright test -c playwright.config.ts ui/workspace-keyboard-flows.test.ts`
- `pnpm exec playwright test -c playwright.config.ts ui/app-restoration.test.ts --grep "retrying a failed run does not duplicate the original user message|chat file links open project files in the workspace and keep trailing punctuation out of hrefs|sending another prompt while a run is active queues it and starts it after the first run finishes"`
- `pnpm exec playwright test -c playwright.config.ts ui/app-manual-edit.test.ts --grep "HTML preview stays rendered after switching from Preview to Code and back"`
- `pnpm exec playwright test -c playwright.config.ts ui/real-daemon-run.test.ts --grep "plugin authoring produces a generated-plugin scaffold with action cards"`
- `pnpm exec playwright test -c playwright.config.ts ui/app.test.ts --grep "sending preview comments keeps the preview live and refreshes it with the follow-up artifact"`
- `pnpm exec playwright test -c playwright.config.ts ui/diagnostics-export.test.ts`
- `pnpm exec playwright test -c playwright.config.ts ui/settings-design-systems.test.ts`
- `pnpm exec playwright test -c playwright.config.ts ui/design-systems-manager.test.ts`
- `pnpm exec playwright test -c playwright.config.ts ui/automations-page.test.ts --grep "places a newly created automation at the top of the list and highlights it|renders the routine target and last-run status in the row summary|keeps saved automations ordered by newest createdAt first"`
- `pnpm exec playwright test -c playwright.config.ts ui/settings-connectors-auth-happy-path.test.ts --grep "disconnecting and reconnecting keeps the connector usable without stale pending state|switches from Connect to Disconnect on success, then returns to Connect after a successful disconnect"`
- `pnpm exec playwright test -c playwright.config.ts ui/settings-connectors-auth-recovery.test.ts`
- `pnpm exec vitest run tests/chat-route.test.ts tests/diagnostics-export.test.ts tests/projects-routes.test.ts tests/runs.test.ts tests/amr-acp-integration.test.ts tests/runtimes/env-and-detection.test.ts tests/runtimes/resolve-model.test.ts`
- `pnpm exec vitest run tests/connection-test.test.ts --testNamePattern "reports an early-phase diagnostics block when the agent CLI is missing \(#2248\)"`
- `pnpm exec vitest run tests/mcp-spawn.test.ts`
- `pnpm exec vitest run tests/runtimes/registry-and-args.test.ts --testNamePattern "codex model picker includes current OpenAI choices in priority order"`

### CODE REVIEW

**Review by lefarcen — COMMENTED** [review]
Hey @AmyShang-alt! The test coverage here is thorough — the breakdown across chat routes, diagnostics export, AMR normalization, auth recovery, and the `startServer` listening-event fix covers a lot of ground in one PR.

One small ask: could you add a `## Surface area` section to the PR body? For a pure test + docs PR like this, it would likely just be **None** checked — but it helps pool reviewers scope the blast radius at a glance without having to diff every file themselves.

Once that's added, this is in good shape for pool review! ✨

**Comment by mrcfps** [comment]
@AmyShang-alt I'm holding off on generating review comments for nexu-io/open-design#3207 because this pull request has merge conflicts right now.

Please resolve the conflicts with main and push the updated branch. Once that's done, request or wait for the review to run again and I'll take another look.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: e2e/lib/fake-agents.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export async function createFakeAgentRuntimes(
       : script;
```
Comment (mrcfps) [inline e2e/lib/fake-agents.ts:85]: Original requested location: e2e/lib/fake-agents.ts RIGHT line 83

On Windows this no longer produces a runnable wrapper. `script` now ends with `.cjs`, but `script.replace(/\.js$/i, .cmd)` leaves `bin` unchanged, so the batch launcher gets written back into the `.cjs` file and `CLAUDE_BIN`/`CODEX_BIN`/… point at that `.cjs` path instead of a real `.cmd` shim. The daemon then tries to exec a batch script through a `.cjs` filename, which breaks every fake-runtime E2E/smoke that uses this helper on Windows. Can we switch the replacement to the new `.cjs` suffix (or derive the wrapper path from `path.parse`) so the JS payload stays in `*.cjs` and the env var points at an actual `*.cmd` wrapper?


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: apps/daemon/src/server.ts [hunk 2]
```diff
@@ -2094,6 +2094,30 @@ function reconcileAssistantMessageOnRunEnd(db, runs, run) {
     });
 }
 
+
+function isPluginAuthoringIntent(input) {
+  const text = [input?.message, input?.currentPrompt]
+    .filter((value) => typeof value === 'string' && value.trim().length > 0)
+    .join('\n');
+  if (!text) return false;
+  return /Create an Open Design plugin for:/i.test(text)
```
Comment (mrcfps) [inline apps/daemon/src/server.ts]: This completion guard only fires for one exact English prompt shape. In the same `/api/runs` flow we already resolve and persist structured plugin metadata onto the run, so a plugin-authoring invocation that comes from CLI/SDK, a reworded shortcut, or a localized prompt will still exit `succeeded` even if `generated-plugin/open-design.json` and `generated-plugin/SKILL.md` were never written. That means the false-positive success you are fixing can come back as soon as the prompt text changes. Can we key this check off structured plugin identity (for example the resolved `pluginId`/snapshot) instead of matching `Create an Open Design plugin for:` in the prompt text?


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by mrcfps — COMMENTED** [review]
@AmyShang-alt Thanks for pushing this regression-coverage sweep — I found two merge-safe follow-ups where the new hardening looks a bit brittle across Windows and non-shortcut plugin-authoring entry points.



<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Review by mrcfps — APPROVED** [review]
@AmyShang-alt I re-checked the latest head and the follow-up addresses the two earlier gaps: the fake-agent Windows wrapper now derives a real `.cmd` shim from the `.cjs` script path, and the plugin-authoring completion guard now keys off structured plugin identity instead of prompt text. I also spot-checked the related AMR normalization and `startServer` listening-event changes against the updated regression coverage, and I do not see any remaining actionable issues. Thanks for tightening this regression sweep. ✨



<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by lefarcen** [comment]
All gates green — flagged for the maintainers. Thanks @AmyShang-alt for pushing through the iteration, and @mrcfps for the thorough review! ✨

**Review by mrcfps — APPROVED** [review]
@AmyShang-alt I re-checked the latest head and spot-checked the substantive daemon changes against the added regression coverage: the plugin-authoring completion guard now keys off structured plugin identity instead of prompt text, the fake-agent Windows wrapper fix is in place, and `startServer` now waits for the actual listening event before reporting its bound URL. I also reviewed the new daemon/UI regression cases around diagnostics export, raw HTML routes, connector auth recovery, automations ordering, and design-system management, and I don’t see any remaining actionable issues. Thanks for tightening up such a broad regression sweep. ✨



<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=reviewer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>
